storage/innobase/xtrabackup/src/xbcloud/azure.h

```diff
@@ -0,0 +1,265 @@
+      Event_handler *h, Azure_client::async_download_callback_t callback,
+      CURLcode rc, const Http_connection *conn, ulong count);
+
+ public:
+  Azure_client(const Http_client *client, const std::string &storage_account,
+               const std::string &access_key, bool development_storage,
+               const ulong max_retries, const ulong max_backoff)
+      : http_client(client),
+        storage_account(storage_account),
+        access_key(access_key),
+        max_retries(max_retries),
+        max_backoff(max_backoff) {
+    if (development_storage) {
+      host = "127.0.0.1:10000/" + storage_account;
+      endpoint = "http://" + host;
+      protocol = Http_request::HTTP;
+    } else {
+      host = storage_account + ".blob.core.windows.net";
+      endpoint = "https://" + host;
+      protocol = Http_request::HTTPS;
+    }
+    rtrim_slashes(host);
+    signer = std::unique_ptr<Azure_signer>(
+        new Azure_signer(storage_account, access_key, development_storage));
+  }
+
+  void set_endpoint(const std::string &ep) {
+    endpoint = ep;
+    if (endpoint.find("https://") == 0) {
+      protocol = Http_request::HTTPS;
+      host = endpoint.substr(8);
+    } else if (endpoint.find("http://") == 0) {
+      protocol = Http_request::HTTP;
+      host = endpoint.substr(7);
+    } else {
+      protocol = Http_request::HTTPS;
+      host = endpoint;
+    }
+    rtrim_slashes(host);
+  }
+
+  bool delete_object(const std::string &container, const std::string &name);
+
+  bool create_container(const std::string &name);
+
+  bool container_exists(const std::string &name, bool &exists);
+
+  bool upload_object(const std::string &container, const std::string &name,
+                     const Http_buffer &contents);
+
+  Http_buffer download_object(const std::string &container,
+                              const std::string &name, bool &success);
+
+  bool async_upload_object(
+      const std::string &container, const std::string &name,
+      const Http_buffer &contents, Event_handler *h,
+      async_upload_callback_t callback = {},
+      const Http_request::headers_t &extra_http_headers = {});
+
+  bool async_download_object(
+      const std::string &container, const std::string &name, Event_handler *h,
+      async_download_callback_t callback = {},
+      const Http_request::headers_t &extra_http_headers = {});
+
+  bool async_delete_object(const std::string &container,
+                           const std::string &name, Event_handler *h,
+                           const async_delete_callback_t callback);
+
+  bool list_objects_with_prefix(const std::string &container,
+                                const std::string &prefix,
+                                std::vector<std::string> &objects);
+
+  ulong get_max_retries() { return max_retries; }
+
+  ulong get_max_backoff() { return max_backoff; }
+
+  void retry_error(Http_response *resp, bool *retry);
```

Comment:
will retry using the logic implemented by @altmannmarcelo ??

Reply:
yes. It will retry the using same logic. just extend the same method.